swiftgen-cli/ArgumentsUtils.swift

```diff
@@ -47,12 +47,20 @@ enum OutputDestination: ArgumentConvertible {
     }
   }
 
-  func write(content: String) {
+  func writeIfChanged(content: String) {
+    write(content, onlyIfChanged: true)
+  }
+  
+  func write(content: String, onlyIfChanged: Bool = false) {
     switch self {
     case .Console:
       print(content)
     case .File(let path):
       do {
+        if try path.isContentEqualTo(content) && onlyIfChanged {
```

Comment:
I think inverting the order of the test (`onlyIfChanged && try path.isContentEqualTo(content)`) would be more efficient, so it doesn't read the file first then don't care about the result if `onlyIfChanged` is `false`, but rather check the `onlyIfChanged` flag first and read the file to compare contents only if needed.